src/pages/index.js

```diff
@@ -380,7 +378,11 @@ const Index = ({ data: initialData }) => {
       </div>
       <div className={classes.centerContainer}>
         <div className={classes.searchBarContainer}>
-          <Logo includeText className={classes.logo} />
+          <Logo
+            includeText
+            color={enableBackgroundImages ? 'white' : null}
+            className={classes.logo}
+          />
           <SearchInput className={classes.searchBar} />
```

Comment:
There's a problem with theming this whole page: the text on white backgrounds is also white, such as the input for the search bar or the demo achievements text. I apologize for not foreseeing this when I encouraged you to make it a page-level theme wrapper.

I'll let you fix this how you think is best but can share my $0.02. In this case, I like your original approach of using the extended theme only for specific sections of the page. Then you could move the HOC logic into the page or convert it into a hook that returns the extended theme. (A standalone hook seems overkill, because we only plan to use that logic in this one instance, so it would probably want to live in the same file with tests running on the page itself.)